nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/repository/schema/RepositoryRecordSchema.java

```diff
@@ -91,4 +97,42 @@
         final UnionRecordField repoUpdateField = new UnionRecordField(REPOSITORY_RECORD_UPDATE_V1, Repetition.EXACTLY_ONE, createOrUpdate, delete, swapOut, swapIn);
         REPOSITORY_RECORD_SCHEMA_V1 = new RecordSchema(Collections.singletonList(repoUpdateField));
     }
+
+    static {
```

Comment:
I believe the only difference between this static block and the other is which FlowFileSchema should be used (FLOWFILE_SCHEMA_V1 vs. FLOWFILE_SCHEMA_V2), correct? This is quite a lot of code to duplicate. Can we instead just have a
`private static RecordSchema createSchema(RecordSchema flowFileSchema)`
type of method, and have it just use the given FlowFileSchema? I believe this would eliminate all of the duplication.